tests/testthat/test-add_data.R

```diff
@@ -88,3 +88,73 @@ test_that("addMarkov throws errors.", {
   expect_error(addMarkov(dd, transMat = mat6, chainLen = 5, wide = TRUE, start0lab = "xy"), class = "simstudy::start0probNotInTransMat")
   
 })
+
+# addSynthetic ----
+
+test_that("addSynthetic throws errors.", {
+  
+  ### Create fake "real" data set
+  
+  d <- defData(varname = "a", formula = 3, variance = 1, dist = "normal")
+  d <- defData(d, varname = "b", formula = 5, dist = "poisson")
+  d <- defData(d, varname = "c", formula = 0.3, dist = "binary")
+  d <- defData(d, varname = "d", formula = "a + b + 3*c", variance = 2, dist = "normal")
+  
+  A <- genData(1000, d, id = "index")
+  
+  def <- defData(varname = "x", formula = 0, variance = 5)
+  
+  S <- genData(120, def)
+  
+  expect_error(ddSynthetic(dtFrom = A))
+  
+  x <- c(1 ,2, 3)
+  expect_error(addSynthetic(dtOld = x, dtFrom = A))
+  expect_error(addSynthetic(dtOld = S, dtFrom = x))
+  expect_error(addSynthetic(dtOld = S, dtFrom = A, id = "index"))
+  expect_error(addSynthetic(dtOld = S, dtFrom = A, id = "id"))
```

Comment:
Also add a check for error message/class here.

Reply:
In addition to the issue mentioned above being explicit about what is `expect`ed here helps readers (i.e. me or future you) understand what the intent of this test is without having to think about it or worse look into the code.

Reply:
Fixed